### PR TITLE
To do (add remove) feature

### PR DESCRIPTION
**Pull request by Rebelzob**
# To-do list Project

- Removed the hardcoded items in the array.
- Created the CRUD file for the functionality.
   
  > Implemented the Adding a new task function.
  > Implemented the function for deleting a task.
  > Implemented the editing function.
  > Implemented the index behavior when adding and updating when deleted.
  > Local storage implemented in all instances.

### CODE REVIEW

File: src/js/crud.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  }
+  // Edits task description, update local storage and the UI.
+  editTask(taskId, newDescription) {
+    const taskObject = this.todoTask.find(task => task.description === taskId);
+    if (taskObject) {
+      taskObject.description = newDescription;
+      this.updateLocalStorage();
+      this.render();
+    }
+  }
```
Comment (Lizdev-05) [inline src/js/crud.js:129]: - [ ] Please from the project requirement, all changes to the To-Do List should be saved in local storage but after editing a task in your to-do app the new changes is not saved in the localStorage as you can see below :point_down: 
  
  [Screencast from 07-14-2023 11:17:31 PM.webm](https://github.com/Rebelzob/To-do-app/assets/73854999/b5dfb0a3-39fb-474c-9fc4-28316afed739)

File: src/js/crud.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    this.updateLocalStorage();
+  }
+  //Adds a new task with the supplied description, updates local storage and the UI.
+  addTask(description) {
+    if (description.trim() !== '') {
+      const newTask = { 
+        description,
+        completed: false,
+        index: this.todoTask.length,
+      };
```
Comment (Lizdev-05) [inline src/js/crud.js:108]: - [ ] Please makes sure that property `index` set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). However as you can see in the image below, your first task has `index` to be `0` which is suppose to be 1 and the second task too to have `index` to be `1` which is suppose to be `2` :blush: 
  
  Screenshot from 2023-07-14 23-20-32

File: src/js/crud.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    this.todoTask.forEach((task, i) => {
+      let taskElement = document.createElement('li');
+      taskElement.classList.add('task-container');
+  
+      let taskContent = document.createElement('div');
+      taskContent.classList.add('task');
+      
+      let checkbox = document.createElement('input');
+      checkbox.type = 'checkbox';
+      checkbox.checked = task.completed;
```
Comment (Lizdev-05) [inline src/js/crud.js:57]: - [ ] Please from the project requirement, all changes to the To-Do List should be saved in local storage so make sure that if a status of a task is change for instance from completed `false` to `true` (by checking the checkbox ) the localStorage is updated. As you can see in the video below if I check the checkbox of a task. the completed is not changing from `false` to `true` :smile: 

[Screencast from 07-14-2023 11:34:34 PM.webm](https://github.com/Rebelzob/To-do-app/assets/73854999/080d2291-a945-4028-84de-592ab940249f)

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @Rebelzob , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
:heavy_check_mark: Professional readme file
:heavy_check_mark: Good commit history
:heavy_check_mark: No linter error

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/crud.js (hunk 3 above)
Comment (Rebelzob) [reply src/js/crud.js:57]: Hi Liz this implementation is meant for the next part of the project, not this one.

image

https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md

**Comment by Rebelzob** [comment]
Please be aware that one of the changes suggested by the reviewer is meant to be implemented as part of the next milestone which is interavtive_list, thanks!
https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md

**Review by Munsa1 — APPROVED** [review]
Hi @Rebelzob, 
Approved 1

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes For This Milestone :heavy_check_mark: 
App Works As Expected :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**